crates/miden-objects/src/batch/proven_batch.rs

```diff
@@ -138,14 +172,15 @@ impl Deserializable for ProvenBatch {
         let output_notes = Vec::<OutputNote>::read_from(source)?;
         let batch_expiration_block_num = BlockNumber::read_from(source)?;
 
-        Ok(Self::new_unchecked(
+        Self::new(
             id,
             reference_block_commitment,
             reference_block_num,
             account_updates,
             input_notes,
             output_notes,
             batch_expiration_block_num,
-        ))
+        )
+        .map_err(|e| DeserializationError::UnknownError(e.to_string()))
```

Comment:
This could potentially swallow source errors, but there's not much we can do. We could create a report but that is quite unwieldy. I opened an issue in winterfell for revisiting the deserialization error https://github.com/facebook/winterfell/issues/374 which would address this more holistically.

Reply:
Thanks for opening the issue. Yes, the current deserialization error doesn't leave us much room to move.